docs/source/parallelism.rst

```diff
@@ -75,5 +75,62 @@ different simulations on the two halves we would write.
 To access the communicator a mesh was created on, we can use the
 ``mesh.comm`` property, or the function ``mesh.mpi_comm()``.
 
+Ensemble parallelism
+=======================
+
+Ensemble parallelism means solving simultaneous copies of the model
+with different coefficients, RHS or initial data, in situations that
+require communication between the copies. Use cases include ensemble
+data assimilation, uncertainty quantification, and time parallelism.
+For large models this will require standard spatial parallelism to
+solve each model copy as well. This requires sub-communicators, even
+if there is only one spatial process per ensemble member, since
+Firedrake does not do anything different for this special case.
+
+Firedrake provides some wrappers to MPI calls so that
+ensemble-member-to-ensemble-member communication can be expressed at
+the level of ``Function``. This is handled by the ``Ensemble``
+class. Instantiating an ensemble requires a communicator (usually
+``MPI_COMM_WORLD``) plus the number of ranks for spatial communication
+within each ensemble subcommunicator (4, in the case of the example
+below). The program should then be executed with ``mpiexec`` as
+normal, specifying the total size of the commicator which should be
+equal to the product of the number of ranks for spatial communication
+and the number of ranks for ensemble communication.
+
+.. code-block:: python
+
+   from firedrake import *
+
+   my_ensemble = Ensemble(COMM_WORLD, 4)
+
+Then, the spatial sub-communicator must be passed to ``Mesh`` (or via
+inbuilt mesh types), so that it will then be used by function spaces
+and functions derived from the mesh.
+
+.. code-block:: python
+
+    x, y = SpatialCoordinate(mesh)
+    V = FunctionSpace(mesh, "CG", 1)
+    u = Function(V)
+
+The ensemble sub-communicator is then available at ``Ensemble.ensemble_comm``.
+.. code-block:: python
+
+    q = Constant(manager.ensemble_comm.rank + 1)
+    u.interpolate(sin(q*pi*x)*cos(q*pi*y))
+
+MPI communications across the spatial sub-communicator (i.e., within
+an ensemble member) are handled automatically by Firedrake, whilst MPI
+communications across the ensemble sub-communicator (i.e., between ensemble
+members) are handled through methods of ``Ensemble``. Currently only
+global reductions are supported.
```

Comment:
I think this is cryptic. I don't think that anyone who doesn't already understand how our ensemble communicators work will understand it after this either.